packages/apollo-server-core/src/ApolloServer.ts

```diff
@@ -357,14 +358,16 @@ export class ApolloServerBase {
     // TODO: This is a bit nasty because the subscription server needs this.schema synchronously, for reasons of backwards compatibility.
     const _schema = this.initSchema();
 
-    if (_schema instanceof GraphQLSchema) {
+    if (isSchema(_schema)) {
       const derivedData = this.generateSchemaDerivedData(_schema);
       this.schema = derivedData.schema;
       this.schemaDerivedData = Promise.resolve(derivedData);
-    } else {
+    } else if (_schema.then) {
       this.schemaDerivedData = _schema.then(schema =>
         this.generateSchemaDerivedData(schema),
       );
+    } else {
+      throw new Error("ApolloServer encountered a programming error. For some reason, we were unable to resolve a valid GraphQLSchema. Please open an issue on the apollo-server repo with a reproduction of this error if possible.");
```

Comment:
Couple of editorial bits, if you don't object:

* I don't think we should refer to Apollo Server by its name, as it sets precedence for doing that in more error messages (which seems repetitive, though I know we do it occasionally?), and it should be determinable by the stack trace anyhow?  Curious what the motivation was though, if I'm missing something!
* I'd suggest removing the `apollo-server` bit from the repo suggestion since it's, on its own, not specific enough to get the user to the right place? (Optionally, could say "file an issue on GitHub"?):

```suggestion
      throw new Error("Unexpected error: Unable to resolve a valid GraphQLSchema.  Please file an issue with a reproduction of this error, if possible.");
```